### ISSUE TITLE
<br> tags are converted to p tags

### ISSUE THREAD

**Issue by CharlieGreenman** [quilljs/quill#4194]
This is an old issue: https://github.com/quilljs/quill/issues/3065

It was closed as a result of quill 2. Just want to say still an issue and that we should look into. Thank you

**Comment by kozi** [comment]
Is there currently a good workaround for this?

**Comment by CharlieGreenman** [comment]
> Is there currently a good workaround for this?

I am using tiptap instead

**Comment by kozi** [comment]
> > Is there currently a good workaround for this?
> 
> I am using tiptap instead

But this is only a "core"-library where I still have to do *some* things myself, right?

Basically, I think the quill editor is good.
However, I wonder how such an essential functionality as `<br>` or `SHIFT+RETURN` is not supported.

**Comment by CharlieGreenman** [comment]
Nope, with tiptap more functionality than quill. I've already solved the br problem with tiptap via importing one of the tiptap extensions.

**Comment by Napam** [comment]
I experience issues with `<br>` not showing up as well.

In quill 1.3 i used the `quill.root.innerHTML` to store the quill value (the HTML would be used in other places, so using the Delta format was just inconvenient). But I can't now in quill 2 since the lists aren't semantic HTML anymore, so now I use `quill.getSemanticHTML()`, but that function  removes the `<br>` tags.

**Comment by YakovcevMark** [comment]
> I experience issues with `<br>` not showing up as well.
> 
> In quill 1.3 i used the `quill.root.innerHTML` to store the quill value (the HTML would be used in other places, so using the Delta format was just inconvenient). But I can't now in quill 2 since the lists aren't semantic HTML anymore, so now I use `quill.getSemanticHTML()`, but that function removes the `<br>` tags.

Yes, this problem still persists.
I don't want to use any other library, so I wrote a handler to add a `<br>` tag to empty `<p>` tags.
This code may not cover all cases, but the test version looks something like this:

`  quill.on('text-change', () => {
        let htmlForOnChange = '';
        const semanticHtml = quill.getSemanticHTML();
        let indexOfStartPTag = semanticHtml.indexOf('<p>');
        let nexPTag = -1;
        let endPTag = -1;
        if (~indexOfStartPTag) {
          htmlForOnChange += semanticHtml.slice(0, indexOfStartPTag);

          while (~indexOfStartPTag) {
            endPTag = semanticHtml.indexOf('</p>', indexOfStartPTag);
            const subStr = semanticHtml.slice(indexOfStartPTag, endPTag + 4);
            
            if (subStr === '<p></p>') {
              htmlForOnChange += '<p><br/></p>';
            } else {
              htmlForOnChange += semanticHtml.slice(indexOfStartPTag, endPTag + 4);
            }

            nexPTag = semanticHtml.indexOf('<p>', endPTag);

            if (~nexPTag) {
              htmlForOnChange += semanticHtml.slice(endPTag + 4, nexPTag);
            } else {
              htmlForOnChange += semanticHtml.slice(endPTag + 4);
            }

            indexOfStartPTag = nexPTag;
          }
        } else {
          htmlForOnChange = semanticHtml;
        }
        
        onChange(htmlForOnChange);
      });`

**Comment by manonlespes** [comment]
@YakovcevMark thanks for tips, I did a few changes to make it work on my code but it is a nice work around.

Note : `<br>` is problematic for people who navigate with the aid of screen reading technology, as screen readers may announce the presence of the element, but not any content contained within `<br>`s.
Thus, since in my case line breaks are purely aesthetic, I added an `aria-hidden` on the tag `<p>` :)
